### PR TITLE
tftui: init at v0.8

### PR DESCRIPTION
**Pull request by srounce**
## Description of changes

Added `tftui` package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aldoborrero** [comment]
@srounce whilst reviewing:

```
framework :: Dev/aldoborrero/nixpkgs ‹feat/llm-support› % nr nixpkgs#nixpkgs-review pr 265844
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/265844/head:refs/nixpkgs-review/1
remote: Enumerating objects: 9879, done.
remote: Counting objects: 100% (4658/4658), done.
remote: Compressing objects: 100% (90/90), done.
remote: Total 9879 (delta 4578), reused 4608 (delta 4560), pack-reused 5221
Receiving objects: 100% (9879/9879), 6.54 MiB | 547.00 KiB/s, done.
Resolving deltas: 100% (6788/6788), completed with 1240 local objects.
From https://github.com/NixOS/nixpkgs
 * [new branch]                master                -> refs/nixpkgs-review/0
 * [new ref]                   refs/pull/265844/head -> refs/nixpkgs-review/1
$ git worktree add /home/aldo/.cache/nixpkgs-review/pr-265844/nixpkgs 8ab2f09522d5aefabf1100b3f59a3bde628cb2b4
Preparing worktree (detached HEAD 8ab2f09522d5)
Updating files: 100% (37907/37907), done.
HEAD is now at 8ab2f09522d5 nixos/qemu-vm: fix infinite recursion
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/aldo/.cache/nixpkgs-review/pr-265844/nixpkgs nixpkgs-overlays=/tmp/tmpjdis8uzz -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git merge --no-commit --no-ff dbbd2184b46bee5815c675eafed9a6d95f15fc39
Automatic merge went well; stopped before committing as requested
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/aldo/.cache/nixpkgs-review/pr-265844/nixpkgs nixpkgs-overlays=/tmp/tmpjdis8uzz -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
1 package added:
tftui-v0.8

$ nom build --nix-path nixpkgs=/home/aldo/.cache/nixpkgs-review/pr-265844/nixpkgs nixpkgs-overlays=/tmp/tmpjdis8uzz --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed -f /home/aldo/.cache/nixpkgs-review/pr-265844/build.nix
these 3 derivations will be built:
  /nix/store/34cnhzjqms08rca92l6qq6xn0crm3hzn-source.drv
  /nix/store/5hz28hq3g4hpxj4a9rl5pbj9p1n8852m-tftui-v0.8.drv
  /nix/store/d02mhg907dw1dcl7y8m57grszp99rjjb-review-shell.drv
these 7 paths will be fetched (1.44 MiB download, 10.79 MiB unpacked):
  /nix/store/lg7m7x61dfnjmmc7j3z2hzpcsycw8015-python3.11-backoff-2.2.1
  /nix/store/lr2215vaw8c99i4hhrm5xg9fqw47vb2d-python3.11-importlib-metadata-6.8.0
  /nix/store/qbamz2rpcs0ihzix3c8fqhk3819pfhyi-python3.11-monotonic-1.6
  /nix/store/6v1lzbl26havc5iyz7x0z3x5v4y806ny-python3.11-posthog-3.0.2
  /nix/store/mgbg4fw176sj09lrcw85z914rhgihpwk-python3.11-rich-13.5.2
  /nix/store/7pxy5nr0001bj8a0c10zk7k4kw2inz97-python3.11-textual-0.40.0
  /nix/store/73mygpxxl95cxk82m64iy1hrdlgg32zc-python3.11-zipp-3.16.2
copying path '/nix/store/lg7m7x61dfnjmmc7j3z2hzpcsycw8015-python3.11-backoff-2.2.1' from 'https://cache.nixos.org'
copying path '/nix/store/qbamz2rpcs0ihzix3c8fqhk3819pfhyi-python3.11-monotonic-1.6' from 'https://cache.nixos.org'
copying path '/nix/store/mgbg4fw176sj09lrcw85z914rhgihpwk-python3.11-rich-13.5.2' from 'https://cache.nixos.org'
copying path '/nix/store/73mygpxxl95cxk82m64iy1hrdlgg32zc-python3.11-zipp-3.16.2' from 'https://cache.nixos.org'
source> building '/nix/store/34cnhzjqms08rca92l6qq6xn0crm3hzn-source.drv'
copying path '/nix/store/lr2215vaw8c99i4hhrm5xg9fqw47vb2d-python3.11-importlib-metadata-6.8.0' from 'https://cache.nixos.org'
copying path '/nix/store/6v1lzbl26havc5iyz7x0z3x5v4y806ny-python3.11-posthog-3.0.2' from 'https://cache.nixos.org'
copying path '/nix/store/7pxy5nr0001bj8a0c10zk7k4kw2inz97-python3.11-textual-0.40.0' from 'https://cache.nixos.org'
source>
source> trying https://github.com/idoavrah/terraform-tui/archive/refs/tags/vv0.8.tar.gz
source>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
source>                                  Dload  Upload   Total   Spent    Left  Speed
source>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
source>   0    14    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
source> curl: (22) The requested URL returned error: 404
source> error: cannot download source from any mirror
error: builder for '/nix/store/34cnhzjqms08rca92l6qq6xn0crm3hzn-source.drv' failed with exit code 1;
       last 8 log lines:
       >
       > trying https://github.com/idoavrah/terraform-tui/archive/refs/tags/vv0.8.tar.gz
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       >   0    14    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
       > curl: (22) The requested URL returned error: 404
       > error: cannot download source from any mirror
       For full logs, run 'nix log /nix/store/34cnhzjqms08rca92l6qq6xn0crm3hzn-source.drv'.
error: 1 dependencies of derivation '/nix/store/5hz28hq3g4hpxj4a9rl5pbj9p1n8852m-tftui-v0.8.drv' failed to build
error: 1 dependencies of derivation '/nix/store/d02mhg907dw1dcl7y8m57grszp99rjjb-review-shell.drv' failed to build
┏━ 1 Errors:
┃ error: builder for '/nix/store/34cnhzjqms08rca92l6qq6xn0crm3hzn-source.drv' failed with exit code 1;
┃        last 8 log lines:
┃        >
┃        > trying https://github.com/idoavrah/terraform-tui/archive/refs/tags/vv0.8.tar.gz
┃        >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
┃        >                                  Dload  Upload   Total   Spent    Left  Speed
┃        >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
┃        >   0    14    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
┃        > curl: (22) The requested URL returned error: 404
┃        > error: cannot download source from any mirror
┃        For full logs, run 'nix log /nix/store/34cnhzjqms08rca92l6qq6xn0crm3hzn-source.drv'.
┣━ Dependency Graph:
┃       ┌─ ↓︎ ✔︎ python3.11-backoff-2.2.1
┃       ├─ ↓︎ ✔︎ python3.11-monotonic-1.6
┃    ┌─ ↓︎ ✔︎ python3.11-posthog-3.0.2
┃    │  ┌─ ↓︎ ✔︎ python3.11-rich-13.5.2
┃    │  │  ┌─ ↓︎ ✔︎ python3.11-zipp-3.16.2
┃    │  ├─ ↓︎ ✔︎ python3.11-importlib-metadata-6.8.0
┃    ├─ ↓︎ ✔︎ python3.11-textual-0.40.0
┃    ├─ ⚠︎ source failed with exit code 1 after ⏱︎ 1s
┃ ┌─ ⏳︎ tftui-v0.8
┃ ⏳︎ review-shell
┣━━━ Builds           │ Downloads        │ Host
┃        │     │      │     │     │      │ localhost
┃        │     │      │     │ ↓︎ 7 │      │ https://cache.nixos.org
┗━ ∑︎ ⏵︎ 0 │ ✔︎ 0 │ ⏳︎ 2 │ ↓︎ 0 │ ↓︎ 7 │ ⏳︎ 0 │ ⚠︎ Exited after 1 build failures at 14:04:16 after 6s
```

**Review by aldoborrero — APPROVED** [review]
LGTM

File: pkgs/applications/networking/cluster/tftui/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    posthog
+    requests
+    textual
+  ];
+
+  meta = with lib; {
+    description = "TFTUI is a powerful textual GUI that empowers users to effortlessly view and interact with their Terraform state.";
```
Comment (MikaelFangel) [inline pkgs/applications/networking/cluster/tftui/default.nix:39]:
```suggestion
    description = "A powerful textual GUI that empowers users to effortlessly view and interact with their Terraform state";
```

File: pkgs/applications/networking/cluster/tftui/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+in
+python3.pkgs.buildPythonApplication rec {
+  inherit version;
+  pname = "tftui";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "idoavrah";
+    repo = "terraform-tui";
+    rev = "refs/tags/${tag}";
```
Comment (MikaelFangel) [inline pkgs/applications/networking/cluster/tftui/default.nix:18]:
```suggestion
python3.pkgs.buildPythonApplication rec {
  pname = "tftui";
  version = "0.8";
  pyproject = true;

  src = fetchFromGitHub {
    owner = "idoavrah";
    repo = "terraform-tui";
    rev = "v${version}";
```

**Review by MikaelFangel — CHANGES_REQUESTED** [review]
The package should be under the `pkgs/by-name/so/some-example` hierarchy, as it is the preferred way of adding new packages. (:

**Comment by idoavrah** [comment]
Any news on this? Something I can do to help?

**Comment by srounce** [comment]
Sorry I've been sick for the past week or so, gonna revisit and update this to incorporate the feedback tomorrow.

**Pull request closed without merging**